src/otk_external_osbuild/command/gen_depsolve_dnf4.py

```diff
@@ -0,0 +1,101 @@
+import hashlib
+import json
+import os
+import subprocess
+import sys
+
+
+def transform(packages):
+    """Transform the output of `osbuild-depsolve-dnf4` to the output format
+    expected of this external."""
+
+    data = {"tree": {"const": {"internal": {}}}}
+
+    # Expose the direct output as internal-only data, this can be used by
+    # other externals.
+    data["tree"]["const"]["internal"]["packages"] = packages
+
+    # We also store all resolved packages and some meta information about
+    # them, this just turns the list into a more user-friendly accessible
+    # map keyed by package name.
+    data["tree"]["const"]["versions"] = {}
+
+    for package in packages:
+        data["tree"]["const"]["versions"][package["name"]] = package
+
+    return data
+
+
+def mockdata(packages):
+    """Mockdata as used by tests, we don't actually execute the depsolver
+    but return a map that's similar enough in format that the rest of the
+    compile can continue."""
+
+    return [
+        {
+            "name": p,
+            "checksum": "sha256:" + hashlib.sha256(p.encode()).hexdigest(),
+            "remote_location": f"https://example.com/repo/packages/{p}",
+            "version": "",
+            "epoch": "",
+            "release": "",
+            "arch": "",
+        }
+        for p in packages
+    ]
+
+
+def root():
+    data = json.loads(sys.stdin.read())
+    tree = data["tree"]
+    mock = "OTK_UNDER_TEST" in os.environ
```

Comment:
Any downside of writing this slightly compacter into:
```python
    if "OTK_UNDER_TEST" in os.environ:
        # When we are under test we don't call the depsolver at all and instead
        # return a mocked list of things
        packages = mockdata(tree["packages"]["include"])
        ...
``` 
(i.e. `mock` is assigned but never used again, only is this diff, by moving it all into the same line it feels easier to read as there is less context to care about)

Reply:
No reason to, I'll adjust!